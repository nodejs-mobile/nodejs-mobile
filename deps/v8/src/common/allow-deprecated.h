// Copyright 2022 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#ifndef V8_COMMON_ALLOW_DEPRECATED_H_
#define V8_COMMON_ALLOW_DEPRECATED_H_

#if defined(V8_IMMINENT_DEPRECATION_WARNINGS) || \
    defined(V8_DEPRECATION_WARNINGS)

#if defined(V8_CC_MSVC)

#define START_ALLOW_USE_DEPRECATED() \
  __pragma(warning(push)) __pragma(warning(disable : 4996))

#define END_ALLOW_USE_DEPRECATED() __pragma(warning(pop))

#else  // !defined(V8_CC_MSVC)

#define START_ALLOW_USE_DEPRECATED() \
  _Pragma("GCC diagnostic push")     \
      _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")

#define END_ALLOW_USE_DEPRECATED() _Pragma("GCC diagnostic pop")

#endif  // !defined(V8_CC_MSVC)

#else  // !(defined(V8_IMMINENT_DEPRECATION_WARNINGS) ||
       // defined(V8_DEPRECATION_WARNINGS))

#define START_ALLOW_USE_DEPRECATED()
#define END_ALLOW_USE_DEPRECATED()

#endif  // !(defined(V8_IMMINENT_DEPRECATION_WARNINGS) ||
        // defined(V8_DEPRECATION_WARNINGS))

#endif  // V8_COMMON_ALLOW_DEPRECATED_H_
