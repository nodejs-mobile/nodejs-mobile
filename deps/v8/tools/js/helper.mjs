// Copyright 2021 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

export const KB = 1024;
export const MB = KB * KB;
export const GB = MB * KB;
export const kMillis2Seconds = 1 / 1000;
export const kMicro2Milli = 1 / 1000;

export function formatBytes(bytes, digits = 2) {
  const units = ['B', 'KiB', 'MiB', 'GiB'];
  const divisor = 1024;
  let index = 0;
  while (index < units.length && bytes >= divisor) {
    index++;
    bytes /= divisor;
  }
  return bytes.toFixed(digits) + units[index];
}

export function formatMicroSeconds(micro) {
  return (micro * kMicro2Milli).toFixed(1) + 'ms';
}

export function formatDurationMicros(micros, secondsDigits = 3) {
  return formatDurationMillis(micros * kMicro2Milli, secondsDigits);
}

export function formatDurationMillis(millis, secondsDigits = 3) {
  if (millis < 1000) {
    if (millis < 1) {
      return (millis / kMicro2Milli).toFixed(1) + 'ns';
    }
    return millis.toFixed(2) + 'ms';
  }
  let seconds = millis / 1000;
  const hours = Math.floor(seconds / 3600);
  const minutes = Math.floor((seconds % 3600) / 60);
  seconds = seconds % 60;
  let buffer = '';
  if (hours > 0) buffer += hours + 'h ';
  if (hours > 0 || minutes > 0) buffer += minutes + 'm ';
  buffer += seconds.toFixed(secondsDigits) + 's';
  return buffer;
}

// Get the offset in the 4GB virtual memory cage.
export function calcOffsetInVMCage(address) {
  let mask = (1n << 32n) - 1n;
  let ret = Number(address & mask);
  return ret;
}

export function delay(time) {
  return new Promise(resolver => setTimeout(resolver, time));
}

export function defer() {
  let resolve_func, reject_func;
  const p = new Promise((resolve, reject) => {
    resolve_func = resolve;
    reject_func = resolve;
  });
  p.resolve = resolve_func;
  p.reject = reject_func;
  return p;
}
