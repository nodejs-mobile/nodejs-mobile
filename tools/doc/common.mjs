import yaml from 'js-yaml';

export function isYAMLBlock(text) {
  return /^<!-- YAML/.test(text);
}

export function isSourceLink(text) {
  return /^<!-- source_link=([^\s/]+\/)+\w+\.\w+ -->/.test(text);
}

export function arrify(value) {
  return Array.isArray(value) ? value : [value];
}

export function extractAndParseYAML(text) {
  text = text.trim()
             .replace(/^<!-- YAML/, '')
             .replace(/-->$/, '');

  // js-yaml.load() throws on error.
  const meta = yaml.load(text);

  if (meta.added) {
    // Since semver-minors can trickle down to previous major versions,
    // features may have been added in multiple versions.
    meta.added = arrify(meta.added);
  }

  if (meta.napiVersion) {
    meta.napiVersion = arrify(meta.napiVersion);
  }

  if (meta.deprecated) {
    // Treat deprecated like added for consistency.
    meta.deprecated = arrify(meta.deprecated);
  }

  if (meta.removed) {
    meta.removed = arrify(meta.removed);
  }

  meta.changes = meta.changes || [];

  return meta;
}
