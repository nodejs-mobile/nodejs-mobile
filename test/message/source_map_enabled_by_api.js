'use strict';
require('../common');

process.setSourceMapsEnabled(true);

try {
  require('../fixtures/source-map/enclosing-call-site-min.js');
} catch (e) {
  console.log(e);
}

delete require.cache[require
  .resolve('../fixtures/source-map/enclosing-call-site-min.js')];

process.setSourceMapsEnabled(false);

require('../fixtures/source-map/enclosing-call-site-min.js');
