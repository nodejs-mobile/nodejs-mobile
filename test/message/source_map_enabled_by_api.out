*enclosing-call-site.js:16
      throw new Error('an error!')
                ^

Error: an error!
    at functionD (*enclosing-call-site.js:16:17)
    at functionC (*enclosing-call-site.js:10:3)
    at functionB (*enclosing-call-site.js:6:3)
    at functionA (*enclosing-call-site.js:2:3)
    at Object.<anonymous> (*enclosing-call-site.js:24:3)
    at Module._compile (node:internal/modules/cjs/loader:*)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:*)
    at Module.load (node:internal/modules/cjs/loader:*)
    at Function.Module._load (node:internal/modules/cjs/loader:*)
    at Module.require (node:internal/modules/cjs/loader:*)
*enclosing-call-site-min.js:1
var functionA=function(){functionB()};function functionB(){functionC()}var functionC=function(){functionD()},functionD=function(){if(0<Math.random())throw Error("an error!");},thrower=functionA;try{functionA()}catch(a){throw a;};
                                                                                                                                                                                                                           ^

Error: an error!
    at functionD (*enclosing-call-site-min.js:1:156)
    at functionC (*enclosing-call-site-min.js:1:97)
    at functionB (*enclosing-call-site-min.js:1:60)
    at functionA (*enclosing-call-site-min.js:1:26)
    at Object.<anonymous> (*enclosing-call-site-min.js:1:199)
    at Module._compile (node:internal/modules/cjs/loader:*)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:*)
    at Module.load (node:internal/modules/cjs/loader:*)
    at Function.Module._load (node:internal/modules/cjs/loader:*)
    at Module.require (node:internal/modules/cjs/loader:*)
